Frontend/src/test/pages/dashboard/components/Customer/SavedAddresses.test.jsx

```
@@ -0,0 +1,120 @@
+
+// --- Stub AddressCard ---
+vi.mock(
+  '../../../../../pages/dashboard/components/Customer/AddressCard',
+  () => ({
+    AddressCard: ({ address, onEdit, onDelete }) => (
+      <div data-testid={`address-${address.id}`}>
+        <span>{address.name}</span>
+        <button
+          data-testid={`edit-${address.id}`}
+          onClick={() =>
+            onEdit({ ...address, name: address.name + ' Edited' })
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)